### ISSUE TITLE
Build a cookie-cutter template for JSON Transformation packages

### ISSUE THREAD

**Issue by bpeterso2000** [json-transformations/json-transformations#1]
Since JSON Transformations has lots of packages, a cookie-cutter template would be helpful for consistency so that everyone is using the same baseline for the packages they are working on.

**Comment by trp07** [comment]
Sounds good.  I'll work on this.  Should I plan to create a separate json-transformation-cookiecutter repo?  I'll push a candidate cookiecutter soon, then we can iterate on it a bit.

**Comment by trp07** [comment]
or maybe the template should just go into this repo?

**Comment by bpeterso2000** [comment]
Yes, I think a separate repo makes the most sense. Thanks!

**Comment by trp07** [comment]
Hello,

Please take a look at the first attempt when you have a chance:
https://github.com/json-transformations/json-transformations-cookiecutter 

Some questions:
1.  The repo I modelled after used tox.  Thoughts about using tox?  
2.  I noticed in your jsoncut setup.py file, it has extras_requires 'colorama'.  I didn't see that in the source code.  Thoughts?  Should that be there?
3.  Should each of these repos be pushed into PyPI?
4.  I've never done continuous integration.  Should that be included?  i.e. Travis?

Thanks!

**Comment by bpeterso2000** [comment]
1. Yes on Tox.
2. Colorama is used by list keys to color the numbers.  It uses Click echo which will color it only if colorama is installed.
3. Which repos?
4. Yes on Travis-CI

**Comment by bpeterso2000** [comment]
A good reference for tox & Travis is jsonconfig everything's setup there.

**Comment by trp07** [comment]
Ok, great.  I'll take a look at jsonconfig for those.  The repos I was talking about would be those like jsonflatten.  Or should they not be sent to PyPI since they may be integrated into jsoncut later?

**Comment by trp07** [comment]
I'll incorporate the travisCI and tox setup you have in jsonconfig.  Any thoughts on the current setup.py file in the cookiecutter template and what should be added, removed, or changed?
Thanks!

**Comment by bpeterso2000** [comment]
Yes, jsonflatten should be registered with PyPi once it's been reviewed and everything's ready to go.
